src/test/java/com/artipie/asto/S3StorageTest.java

```diff
@@ -92,25 +96,27 @@ void shouldUploadObjectWhenSave(final AmazonS3 client) throws Exception {
     }
 
     @Test
+    @Timeout(5)
     void shouldUploadObjectWhenSaveContentOfUnknownSize(final AmazonS3 client) throws Exception {
         final byte[] data = "data?".getBytes();
         final String key = "unknown/size";
         this.storage().save(
             new Key.From(key),
-            new Content.From(Flowable.just(ByteBuffer.wrap(data)))
+            new Content.From(new OneOffPublisher(ByteBuffer.wrap(data)))
         ).join();
         MatcherAssert.assertThat(this.download(client, key), Matchers.equalTo(data));
     }
 
     @Test
+    @Timeout(15)
```

Comment:
@olegmoz why do we need timeout here? test doesn't hang anymore using `OneOffPublisher`

Reply:
@Vatavuk that's because the test would hang (future never completes) with OneOffPublisher without caching data. That is why I have added timeouts.

Reply:
@olegmoz Yes, but potentially lot of tests might hang if something in the code changes:D, ok let's leave it as a future safety net for this particular scenario.